tracer/src/Datadog.Trace/Util/HexString.cs

```diff
@@ -86,6 +127,37 @@ public static string ToHexString(ulong value, bool lowerCase = true)
         return ToHexString(bytes, lowerCase);
     }
 
+    /// <summary>
+    /// Converts the specified <see cref="TraceId"/> value into a hexadecimal string.
+    /// </summary>
+    [Pure]
+    public static string ToHexString(TraceId value, bool pad16To32 = true, bool lowerCase = true)
+    {
+        if (!pad16To32 && value.Upper == 0)
+        {
+            // trace id fits in 16 hex characters
+            return ToHexString(value.Lower);
+        }
+
+        value = ReverseIfLittleEndian(value);
+        var upper = value.Upper;
+        var lower = value.Lower;
+
+#if NETCOREAPP3_1_OR_GREATER
+        // NOTE: don't use MemoryMarshal.Write(bytes, ref value) because .NET will
```

Comment:
Maybe I'm misunderstanding the comment here, but it looks like `MemoryMarshal.Write` is used below?

Reply:
Note the arguments. We can't pass `value` as the second argument, so instead we write `upper` and `lower` in two separate steps.